src/scipp/core/bins.py

```diff
@@ -108,6 +108,12 @@ def lookup(
     if dim is None:
         dim = func.dim
     func = DataArray(func.data, coords={dim: func.coords[dim]}, masks=func.masks)
```

Comment:
Do we need to transpose the mask as well or is this handled differently in the C++ code?